bookkeeper-server/src/main/java/org/apache/bookkeeper/proto/BookieProtoEncoding.java

```diff
@@ -181,6 +204,21 @@ public Object decode(ByteBuf packet)
                 } else {
                     return BookieProtocol.ReadRequest.create(version, ledgerId, entryId, flags, null);
                 }
+            case BookieProtocol.BATCH_READ_ENTRY:
+                ledgerId = packet.readLong();
+                entryId = packet.readLong();
+                long requestId = packet.readLong();
+                int maxCount = packet.readInt();
+                long maxSize = packet.readLong();
+                if ((flags & BookieProtocol.FLAG_DO_FENCING) == BookieProtocol.FLAG_DO_FENCING
```

Comment:
FENCING is a special reading that will fence the ledger, and no more write allowed; will it allow in the batch read?

Reply:
If we move the recovery to use batch reads then this may be useful